data/transactions/logic/eval_test.go

```diff
@@ -3759,44 +3769,79 @@ func BenchmarkByteLogic(b *testing.B) {
-			"int 1"},
-		{"b/big", "", // u256 / u128 (half sized divisor seems pessimal)
-			`byte 0xa123457601234576012345760123457601234576012345760123457601234576
-			 byte 0x34576012345760123457601234576312; b/; pop`,
-			"int 1"},
-		{"b%big", "", // u256 / u128 (half sized divisor seems pessimal)
-			`byte 0xa123457601234576012345760123457601234576012345760123457601234576
-			 byte 0x34576012345760123457601234576312; b/; pop`,
-			"int 1"},
-		{"bsqrt-big", "",
-			`byte 0xa123457601234576012345760123457601234576012345760123457601234576
-			 bsqrt; pop`,
-			"int 1"},
+		{"bytec", u128 + "pop"},
+
+		{"b+ 128", u128 + u128 + "b+; pop"},
+		{"b- 128", u128 + u128 + "b-; pop"},
+		{"b* 128", u128 + u128 + "b*; pop"},
+		// half sized divisor seems pessimal for / and %
+		{"b/ 128", u128 + u64 + "b/; pop"},
+		{"b% 128", u128 + u64 + "b%; pop"},
+		{"bsqrt 128", u128 + "bsqrt; pop"},
+
+		{"b+ 256", u256 + u256 + "b+; pop"},
+		{"b- 256", u256 + u256 + "b-; pop"},
+		{"b* 256", u256 + u256 + "b*; pop"},
+		{"b/ 256", u256 + u128 + "b/; pop"},
+		{"b% 256", u256 + u128 + "b%; pop"},
+		{"bsqrt 256", u256 + "bsqrt; pop"},
+
+		{"b+ 512", u512 + u512 + "b+; pop"},
+		{"b- 512", u512 + u512 + "b-; pop"},
+		{"b* 512", u512 + u512 + "b*; pop"},
+		{"b/ 512", u512 + u256 + "b/; pop"},
+		{"b% 512", u512 + u256 + "b%; pop"},
+		{"bsqrt 512", u512 + "bsqrt; pop"},
+
+		{"bytec recheck", u128 + "pop"},
 	}
 	for _, bench := range benches {
 		b.Run(bench[0], func(b *testing.B) {
-			benchmarkOperation(b, bench[1], bench[2], bench[3])
+			b.ReportAllocs()
+			benchmarkOperation(b, "", bench[1], "int 1")
+		})
+	}
+}
+
+func BenchmarkByteCompare(b *testing.B) {
+	u64 := "byte 0x8090a0b0c0d0e0f0;"
+	hex128 := "102030405060708090a0b0c0d0e0f000"
+	u128 := "byte 0x" + strings.Repeat(hex128, 1) + ";"
+	u256 := "byte 0x" + strings.Repeat(hex128, 2) + ";"
+	u512 := "byte 0x" + strings.Repeat(hex128, 4) + ";"
+	//u4k := "byte 0x" + strings.Repeat(hex128, 256) + ";"
+
+	benches := [][]string{
+		{"b== 64", u64 + u64 + "b==; pop"},
+		{"b< 64", u64 + u64 + "b<; pop"},
+		{"b<= 64", u64 + u64 + "b<=; pop"},
+		{"b== 128", u128 + u128 + "b==; pop"},
+		{"b< 128", u128 + u128 + "b<; pop"},
+		{"b<= 128", u128 + u128 + "b<=; pop"},
+		{"b== 256", u256 + u256 + "b==; pop"},
+		{"b< 256", u256 + u256 + "b<; pop"},
+		{"b<= 256", u256 + u256 + "b<=; pop"},
+		{"b== 512", u512 + u512 + "b==; pop"},
+		{"b< 512", u512 + u512 + "b<; pop"},
+		{"b<= 512", u512 + u512 + "b<=; pop"},
+		// These can only be run with the maxByteMathSize check removed. They
+		// show that we can remove that check in a later AVM version, as there
+		// is no appreciable cost to even a 4k compare.
+		// {"b== 4k", u4k + u4k + "b==; pop"},
+		// {"b< 4k", u4k + u4k + "b<; pop"},
+		// {"b<= 4k", u4k + u4k + "b<=; pop"},
+	}
+	for _, bench := range benches {
+		b.Run(bench[0], func(b *testing.B) {
+			b.ReportAllocs()
+			benchmarkOperation(b, "", bench[1], "int 1")
```

Comment:
dumb question: so the new benchmark on `b==` and b`<` will not affect cost of these opcodes in earlier protocol versions, and we might update the cost later?

Reply:
Actually, it turns out we were undercharging before. (It's not so serious, because the length is limited to 64 bytes, so the undercharge couldn't be very large.)

So with this change, the cost of these opcodes (which is just 1), is accurate.

Reply:
I see, also read your comments on `maxByteMathSize`, maybe that's what you referring as 64 bytes limit. If we move beyond, e.g., allowing 512 bytes, such that we activate 4k benchmark, will we still be at 1 cost?

Reply:
Right - with this change in place, we could lift the restriction on length for these opcodes, they would still be fast enough. (but that needs a new version.)

Reply:
got it, then it totally makes sense.